_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+| **merge** any branch to `next`                        | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** some commits from `next` branch to `master` | Get the last release for commits on branch `master` (including the one that come from `next`), analyze commits, increase the version; If the commit associated with the last release found is also present on `next`, make the version available on the `latest` channel, otherwise do a regular release. |
+
+##### Future branches configuration
+
+```json
+{
+  "release": {
+    "branches": ["master", "next"]
+  }
+}
+```
+
+Internally **semantic-release** will determines the type of commit that can be done on each branch, based on the number of future branches defined:
+- `"branches": ["master"]` => Any type of commit can be pushed to `master`
+- `"branches": ["master", "next"]` => `master` will accept `fix` and `feat` commits; `next` will accept any type of commits.
+- `"branches": ["master", "next", "next-major"]` => `master` will accept `fix`; `next` will accept `fix` and `feat` commits; `next-major` will accept any type of commits.
+
+Those rules are meant to prevent inconsistent state in releases in which the same releases line is "forked".
+
+For example with the config `"branches": ["master", "next"]` and:
+- last release from `master` is `1.1.0`
+- last release from `next` is `2.0.0`
+
+Committing a breaking change to `master` would trigger the release of `2.0.0` that would fail as the release already exists.
+
+In addition when the `master` and `next` have the same commit head (after a merging `next` into `master` for example), the first commit done on `next` must be a breaking change.
+Otherwise that would result in a "fork" situation.
+
+For example with the config `"branches": ["master", "next"]` and:
+- last release from `master` is `2.0.0`
+- last release from `next` is `2.0.0`
+
+A `feat` commit on `master` will release version `2.1.0`.
+Committing a `feat` commit on `next` would trigger the release of `2.1.0` that would fail as the release already exists.
+
+With specific channel names:
+
+```json
+{
+  "release": {
+    "branches": [
+      "master",
+      {"branch": "next", "channel": "preview"}
+    ]
+  }
+}
+```
+
+#### LTS branches
+
+Workflow to release legacy versions.
+
+Ideal for modules maintaining legacy versions and doing bug fixes and features only releases.
+
+Branch: `1.x`, `2.x`, `master` (any number of lts branches are supported)
+
+Channel: `1.x`, `2.x`, `latest` if supported by the release target
```

Comment:
Is a channel needed? From the perspective of an npm user, when would I ever use the `1.x` channel, versus just using `^1.0.0`? In that example they seem functionally equivalent, though the use of a semantic version has stronger semantic meaning as the `1.x` channel could just point to version `2.0.0`.

Reply:
What's the use case for the `1.x` dist-tag?

Reply:
Releasing on a dist-tag for user using a legacy version

Reply:
So this may be a moot point:

```
> npm publish --tag 1.x
npm ERR! Tag name must not be a valid SemVer range: 1.x
```

Reply:
Yes I didn't realize that. We gonna have to find a naming convention for legacy channel. Maybe `legacy-1.x`.

Reply:
Or `release-1.x` (I'm still trying to avoid the `legacy` term 😉 )